saas_demo/models/saas_operator.py

```diff
@@ -78,23 +80,32 @@ def update_addons_path(self):
     @api.multi
     def restart_odoo(self):
         if self.is_local():
+            test = self.env['ir.config_parameter'].get_param('test_saas_demo')
+            if test:
+                # no need to restart odoo folder in test mode
+                return
```

Comment:
Check via ``test_cr`` instead

Reply:
You mean `self.env.registry.test_cr`? For some reason `test_cr` is `None` in test mode

Reply:
Do you call cr.commit() somewhere? It could be a reason

Reply:
Yes I am, but before `self.env.cr.commit()` `test_cr` is also `None`

Reply:
what if I check the odoo.tools.config parameter ['test_enable'] instead?

Reply:
You need to figure out with rollbacking after tests. And delete unlink() calls